lib/Sema/CSBindings.cpp

```diff
@@ -66,6 +66,14 @@ ConstraintSystem::determineBestBindings() {
         continue;
 
       for (auto &binding : relatedBindings->getSecond().Bindings) {
+        // We need the binding kind for the potential binding to
+        // either be Exact or Supertypes in order for it to make sense
+        // to add Supertype bindings based on the relationship between
+        // our type variables.
+        if (binding.Kind != AllowedBindingKind::Exact
+            && binding.Kind != AllowedBindingKind::Supertypes)
```

Comment:
Only formatting nit - `clang-format` always leaves operators on at the end of the line, so in this case `&&` wouldn't be moved done.

Reply:
Yes, this isn't formatted the same way that our current formatting clang-format rules would format it. I think that's a good thing :).

My recollection is that there was a thread on this not long ago where there was some support for changing the clang-format rule for this since it's often easier to see what's going on when the operators are at the beginning of the continuation line.

We don't, AFAIK, have a rule about clang-formatting all changes, either (if we did, we'd probably make that a git post-receive hook).

Having said all that, we do have a rule that new code should be formatted consistently with surrounding code and in this case this would be an outlier, so I'll add another commit to fix this.

Reply:
Interesting, always good to know the background!